### ISSUE TITLE
-Wsign-compare and signed container size

### ISSUE THREAD

**Issue by GElliott** [ericniebler/range-v3#871]
Use of `ranges::view::all` with a container type that reports size as a singed variable results in a signed/unsigned comparison. This is a compile-time error for those using `-Wsign-compare`.

I've run into this issue w/ Google protobuf's [RepeatedPtrField](https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.repeated_field#RepeatedPtrField) container, but Qt also uses signed variables to report container size ([example](http://doc.qt.io/qt-5/qvector.html#size)).

[Contrived example](https://godbolt.org/g/VF92Yz)

Output:
```
/opt/compiler-explorer/libs/rangesv3/0.3.6/include/range/v3/view/all.hpp:55:43: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
                     RANGES_ASSERT(size(t) == size(begin(t), end(t)));
/opt/compiler-explorer/libs/rangesv3/0.3.6/include/range/v3/view/all.hpp:55:21: note: in expansion of macro 'RANGES_ASSERT'
                     RANGES_ASSERT(size(t) == size(begin(t), end(t)));
```

**Comment by BRevzin** [comment]
Looks like this was fixed by e743c56c2.

**Comment by ericniebler** [comment]
Confirmed, fixed on trunk.
